src/main/java/org/generationcp/breeding/manager/listmanager/dialog/GenerateStockIDsDialog.java

```diff
@@ -265,14 +265,15 @@ private void showMessageValidationForPrefix() {
 		this.txtSpecifyPrefix.focus();
 	}
 
-	boolean isValidPrefix(String prefix) {
-		String pattern = "^[a-zA-Z]*$";
+	boolean isValidPrefix(final String prefix) {
+		//Or Empty string to match previous validation
+		final String pattern = "^[a-zA-Z0-9]*[a-zA-Z]$|^$";
```

Comment:
Extract the regex as a constant. I think it's better to use the same regex as in BMSAPI: [InventoryCommonValidator](https://github.com/IntegratedBreedingPlatform/BMSAPI/blob/924fe7f2db3680a21c65b5ac790f9665b3612397/src/main/java/org/ibp/api/java/impl/middleware/inventory/common/validator/InventoryCommonValidator.java#L24)

Reply:
Modified the regex to `[a-zA-Z0-9]{0,14}[a-zA-Z]|^$ `